### PR TITLE
Adding Bad Wings Support

### PR DESCRIPTION
**Pull request by jasonhazel**
## Description

Adding initial support for [Bad Wings](https://shop.hazel.cc/products/bad-wings) keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/hazel/bad_wings/info.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  "bootloader": "rp2040",  
+  "matrix_size": {
+    "cols": 8,
+    "rows": 5
+  },  
+  "diode_direction": "COL2ROW",
+  "bootmagic": {
+    "enabled": true,
+    "matrix": [0, 0]
+  },
```
Comment (zvecr) [inline keyboards/hazel/bad_wings/info.json:21]:
```suggestion
```

File: keyboards/hazel/bad_wings/info.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  "features": {
+    "audio": false,
```
Comment (zvecr) [inline keyboards/hazel/bad_wings/info.json:23]:
```suggestion
```

File: keyboards/hazel/bad_wings/info.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    "backlight": false,
+    "bootmagic": true,
+    "command": false,
+    "console": false,
+    "deferred_exec": true,
+    "magic": false,
+    "nkro": false,
+    "rgblight": false,
+    "unicode": false
```
Comment (zvecr) [inline keyboards/hazel/bad_wings/info.json:32]:
```suggestion
    "rgblight": false
```

File: keyboards/hazel/bad_wings/info.json [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    "matrix": [0, 0]
+  },
+  "features": {
+    "audio": false,
+    "backlight": false,
+    "bootmagic": true,
+    "command": false,
+    "console": false,
+    "deferred_exec": true,
+    "magic": false,
```
Comment (zvecr) [inline keyboards/hazel/bad_wings/info.json:29]:
```suggestion
```

File: keyboards/hazel/bad_wings/halconf.h [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2023 Jason Hazel (@jasonhazel)
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#pragma once
+
+#define HAL_USE_SERIAL  FALSE
```
Comment (zvecr) [inline keyboards/hazel/bad_wings/halconf.h:6]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/hazel/bad_wings/matrix.c [hunk 6]
```diff
@@ -0,0 +1,137 @@
+// Copyright 2022 @sadekbaroudi (Sadek Baroudi)
```
Comment (keyboard-magpie) [inline keyboards/hazel/bad_wings/matrix.c:1]: this should probably have your name below sadek's

**Review by keyboard-magpie — APPROVED** [review]

File: keyboards/hazel/bad_wings/matrix.c (hunk 6 above)
Comment (jasonhazel) [reply keyboards/hazel/bad_wings/matrix.c:1]: Updating to include myself.

File: keyboards/hazel/bad_wings/keymaps/default/rules.mk [hunk 7]
```diff
@@ -0,0 +1,8 @@
+GRAVE_ESC_ENABLE		= no
+SPACE_CADET_ENABLE      = no
```
[the remark this answers is not in the document]
Comment (jasonhazel) [reply keyboards/hazel/bad_wings/keymaps/default/rules.mk]: Good catch. They do not need to be disabled.  This was a carry over  from when I was using a ProMicro and trying to save space.

File: keyboards/hazel/bad_wings/matrix.c [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+    // reset the current matrix, as we'll be updating and comparing to the old matrix
+    memset(current_matrix, 0, matrixArraySize);
+
+
+    bool debug_output = false;
+    // Loop through the columns, activating one at a time, and read the rows, and place in the new current_matrix
+    for (int c = 0; c < SHIFTREG_COLS; c++) {
+        if (debug_output) {
+            xprintf("column iteration: %d\n", c);
```
Comment (drashna) [inline keyboards/hazel/bad_wings/matrix.c:114]: dprintf?

File: keyboards/hazel/bad_wings/keymaps/default/rules.mk [hunk 9]
```diff
@@ -0,0 +1,6 @@
+MOUSEKEY_ENABLE			= yes
+EXTRAKEY_ENABLE			= yes
+CAPS_WORD_ENABLE		= yes
+TAP_DANCE_ENABLE		= yes
+TRI_LAYER_ENABLE 		= yes
+POINTING_DEVICE_ENABLE  = yes
```
Comment (drashna) [inline keyboards/hazel/bad_wings/keymaps/default/rules.mk:6]: This should probably be at the kb level.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/hazel/bad_wings/matrix.c (hunk 8 above)
Comment (jasonhazel) [reply keyboards/hazel/bad_wings/matrix.c]: updated to use `dprintf`

File: keyboards/hazel/bad_wings/keymaps/default/rules.mk (hunk 9 above)
Comment (jasonhazel) [reply keyboards/hazel/bad_wings/keymaps/default/rules.mk]: We're talking just `POINTING_DEVICE_ENABLE`, right?  if so, that's been updated.

Comment (jasonhazel) [reply keyboards/hazel/bad_wings/keymaps/default/rules.mk]: I didn't mind updating the default keymap - it was mostly changes I've been wanting to make for a while.<

I've moved `TRI_LAYER_ENABLE` and `CAPS_WORD_ENABLE` to the keyboard level rules.mk. Thanks so much for your feedback!

**Pull request merged**